src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+        System.out.println("Now you have " + list.size() + " task(s) in the list.");
+    }
+
+    /**
+     * Adds a todo item to the user's list.
+     * E.g. 'todo Go running' adds the task
+     *      [T][ ] Go running
+     *      to the user's list.
+     * Throws an exception if empty task description is given.
+     */
+    public static void todo() {
+        String description1 = scan.nextLine();
+        try {
+            EmptyTaskException.validate(description1);
+        } catch(EmptyTaskException ex) {
+            System.out.println("Please give a description for your todo item. Try again!");
+            return;
+        }
+
+        String trimmed1 = description1.trim();
+        Task task1 = new Todo(trimmed1);
+        list.add(task1);
+        System.out.println("OK, I have added the task '" + trimmed1 + "' to your list! :)");
+        System.out.println("Now you have " + list.size() + " task(s) in the list.");
+    }
+
+    /**
+     * Adds an event item to the user's list.
+     * Uses '/' to take in the timing of the event.
+     * E.g. 'event Party /From Friday 8pm /to 11pm' adds the task
+     *      [E][ ] Party (From Friday 8pm to 11pm)
+     *      to the user's list.
+     * Throws an exception if no task description, or no event start or end timing is given.
+     */
+    public static void event() {
+        String fullString1 = scan.nextLine();
+        String[] tokens1 = fullString1.split("/");
+
+        try {
+            EmptyTaskException.timedValidate(tokens1);
+            MissingEventTimingException.validate(tokens1);
+        } catch(EmptyTaskException ex) {
+            System.out.println("Please give a description for your event item. Try again!");
+            return;
+        } catch(MissingEventTimingException ex) {
+            System.out.println("Please give a time period for your event item. Try again! XD");
+            System.out.println("E.g. Type 'event meeting /From Monday 10am /to 12pm' to add the task 'meeting' \n" +
+                    "with a time period 'From Monday 10am to 12pm' to your list.");
+            return;
+        }
+
+        String description2 = tokens1[0];
+        String trimmed2 = description2.trim();
+        String from = tokens1[1];
+        String to = tokens1[2];
+        String toTrimmed = to.trim();
+        Task task2 = new Event(trimmed2, from, toTrimmed);
+        list.add(task2);
+        System.out.println("OK, I have added the task '" + trimmed2 + "' to your list! :)");
+        System.out.println("Now you have " + list.size() + " task(s) in the list.");
+    }
+
+    /**
+     * Displays the user's current todo list.
+     */
+    public static void list() {
+        System.out.println("Here are the items in your list:");
+        int i = 0;
+        int j = 1;
+        while (i < list.size()) {
+            System.out.println(j + ". " + list.get(i).toString());
+            i++;
+            j++;
+        }
```

Comment:
Could this be written better using a for loop instead of using a while loop and 2 variables?